Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -100,6 +103,347 @@ public void Session_Error_ShouldPassCorrectSession()
+                    {
+                        Assert.IsTrue(onBeforeTriggered);
+                        Assert.IsFalse(onAfterResetTriggered);
+                        Assert.IsFalse(manualFallbackTriggered);
+                        manualFallbackTriggered = true;
+                        tcs.TrySetResult(true);
+                    }
+                };
+
+                using var realm = await GetRealmAsync(config);
+
+                GetSession(realm).SimulateClientReset("simulated client reset");
+
+                await tcs.Task;
+
+                Assert.IsTrue(manualFallbackTriggered);
+                Assert.IsTrue(onBeforeTriggered);
+                Assert.IsFalse(onAfterResetTriggered);
+            });
+        }
+
+        [Test]
+        public void Session_ClientReset_ManualRecoveryHandler_When_Exception_OnAfterReset()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var tcs = new TaskCompletionSource<bool>();
+                var onBeforeTriggered = false;
+                var manualFallbackTriggered = false;
+                var onAfterResetTriggered = false;
+                var config = await GetIntegrationConfigAsync();
+                config.ClientResetHandler = new DiscardLocalResetHandler
+                {
+                    OnBeforeReset = (beforeFrozen) =>
+                    {
+                        Assert.IsFalse(onBeforeTriggered);
+                        Assert.IsFalse(onAfterResetTriggered);
+                        Assert.IsFalse(manualFallbackTriggered);
+                        onBeforeTriggered = true;
+                    },
+                    OnAfterReset = (beforeFrozen, after) =>
+                    {
+                        Assert.IsTrue(onBeforeTriggered);
+                        Assert.IsFalse(onAfterResetTriggered);
+                        Assert.IsFalse(manualFallbackTriggered);
+                        onAfterResetTriggered = true;
+                        throw new Exception("Exception thrown in OnAfterReset");
+                    },
+                    ManualResetFallback = (session, err) =>
+                    {
+                        Assert.IsTrue(onBeforeTriggered);
+                        Assert.IsTrue(onAfterResetTriggered);
+                        Assert.IsFalse(manualFallbackTriggered);
+                        manualFallbackTriggered = true;
+                        tcs.TrySetResult(true);
+                    }
+                };
+
+                using var realm = await GetRealmAsync(config);
+
+                GetSession(realm).SimulateClientReset("simulated client reset");
+
+                await tcs.Task;
+
+                Assert.IsTrue(manualFallbackTriggered);
+                Assert.IsTrue(onBeforeTriggered);
+                Assert.IsTrue(onAfterResetTriggered);
+            });
+        }
+
+        [Test]
+        public void Session_SyncError_Handler()
```

Comment:
This name is misleading as it's testing the SyncConfiguration.OnSessionError handler, not Session.Error.